### PR TITLE
WIP: Fourth round of refactoring for Manager

### PR DESCRIPTION
**Pull request by jackgr**
/cc @sparkprime

### CODE REVIEW

**Comment by jackgr** [comment]
Couple of notes about this PR:
* Need to wire up API to take a common.Configuration not a common.Template.
* Need to mock the repo provider and remove dependency on live chart.
* Need to add multi level expansion tests and negative tests.

**Comment by sparkprime** [comment]
FYI I already created common.CreateDeploymentRequest for the manager API, and modified the helm CLI to provide this.

**Comment by sparkprime** [comment]
LGTM

**Comment by jackgr** [comment]
@sparkprime PTAL

File: cmd/manager/manager/expander.go [hunk 1]
```diff
@@ -130,25 +77,42 @@ func (e *expander) ExpandConfiguration(conf *common.Configuration) (*ExpandedCon
...
 
+		// It is a chart, so go fetch it, decompress it and unpack it
 		cbr, _, err := e.repoProvider.GetChartByReference(resource.Type)
 		if err != nil {
 			return nil, err
 		}
 
 		defer cbr.Close()
+
+		// Now, load the charts contents into strings that we can pass to exapnsion
```
Comment (sparkprime) [inline cmd/manager/manager/expander.go]: typo exapnsion

File: cmd/manager/manager/expander.go [hunk 2]
```diff
@@ -159,19 +123,27 @@ func (e *expander) expandConfiguration(conf *common.Configuration) (*ExpandedCon
...
 		}
 
+		// Call ourselves recursively with the list of resources returned by expansion
 		expConf, err := e.expandConfiguration(svcResp)
 		if err != nil {
 			return nil, err
 		}
 
-		// TODO: build up layout hiearchically
+		// Append the reources returned by the recursion to the flat list of resources
```
Comment (sparkprime) [inline cmd/manager/manager/expander.go]: typo reources

**Comment by sparkprime** [comment]
LGTM, up to the ongoing discussion of whether the API should instantiate more than one chart in a single call

**Comment by jackgr** [comment]
Thanks, @sparkprime. Yes, we'll need to decided on that point. For now, the server takes a named list and the client sends it one chart at a time with the chart name as the list name. I suspect that'll change to pass more than and to generate unique names as soon as we get to real end to end scenarios.
